brainfuck/executor.go

```diff
@@ -0,0 +1,54 @@
+package brainfuck
+
+func Interpret(program string) {
+	exec := compile(program)
+	exec.Execute()
+}
+
+func compile(input string) *program {
+	p := newParser()
+	result := newProgram(p.parse(input), newMemory())
+	return result
+}
+
+type Parser struct {
+	stack [][]command
+	ptr   int
+}
+
+func newParser() *Parser {
+	return &Parser{
+		stack: [][]command{[]command{}},
+		ptr:   0,
+	}
+}
+
+func (p *Parser) parse(input string) []command {
+	for _, char := range input {
+		p.parseInstruction(char)
+	}
+	return p.stack[0]
+}
+
+func (p *Parser) parseInstruction(char rune) {
+	switch char {
+	case '+':
+		p.stack[p.ptr] = append(p.stack[p.ptr], Increment{})
+	case '-':
+		p.stack[p.ptr] = append(p.stack[p.ptr], Decrement{})
+	case '>':
+		p.stack[p.ptr] = append(p.stack[p.ptr], MoveForward{})
+	case '<':
+		p.stack[p.ptr] = append(p.stack[p.ptr], MoveBackward{})
+	case '.':
+		p.stack[p.ptr] = append(p.stack[p.ptr], Print{})
+	case '[':
+		p.stack = append(p.stack, []command{})
```

Comment:
why command is private, but the other instructions public?

Reply:
Fixed